aten/src/ATen/native/cuda/Copy.cu

```diff
@@ -72,16 +73,30 @@ void copy_device_to_device(TensorIterator& iter, bool non_blocking) {
         gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return x; });
       });
     } else {
-      if (!same_conj && same_type) {
-        AT_DISPATCH_COMPLEX_TYPES(
-            dtype, "copy_conj_", [&] {
-              gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return std::conj(x); });
-            });
+      if (same_neg) {
+        if (!same_conj && same_type) {
+          AT_DISPATCH_COMPLEX_TYPES(
+              dtype, "copy_conj_", [&] {
+                gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return std::conj(x); });
+              });
+        } else {
+          AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(
+              kHalf, kBool, kBFloat16, dtype, "copy_", [&] {
+                gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return x; });
+              });
+        }
       } else {
-        AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(
-            kHalf, kBool, kBFloat16, dtype, "copy_", [&] {
-              gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return x; });
-            });
+        if (!same_conj && same_type) {
+          AT_DISPATCH_COMPLEX_TYPES(
+              dtype, "copy_conj_", [&] {
+                gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return std::conj(-x); });
+              });
+        } else {
+          AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND3(
+              kHalf, kBool, kBFloat16, dtype, "copy_", [&] {
+                gpu_kernel(iter, [] GPU_LAMBDA(scalar_t x) { return -x; });
+              });
```

Comment:
At some point we gonna have to have a chat about our O(2^n) conversions LOL

Reply:
yeah